util/network-devp2p/src/node_table.rs

```diff
@@ -83,6 +83,7 @@ impl NodeEndpoint {
 		let address = match addr_bytes.len() {
 			4 => Ok(SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(addr_bytes[0], addr_bytes[1], addr_bytes[2], addr_bytes[3]), tcp_port))),
 			16 => unsafe {
+				#[allow(clippy::cast_ptr_alignment)] // TODO: Why is this okay?
```

Comment:
well, I'm not super intimate with this piece of code why it is unsafe in the first place but I think it is because we don't want to copy the bytes to perform the conversion `[u8; 16] -> [u16; 8]`

We could do this in safe rust:
```rust
				let mut addr = [0_u8; 16];
				addr.copy_from_slice(&addr_bytes[0..16]);
				Ok(SocketAddr::V6(SocketAddrV6::new(Ipv6Addr::from(addr), tcp_port, 0, 0)))
```
The unsafe scope could be reduced because only `slice::from_raw_parts` is unsafe.

I'm up making it safe, but let's here what the others have to say //cc @ordian @dvdplm

Reply:
AFAICT it is not used on any hot paths (`on_ping`, `on_pong` and `on_neighbours`). It is called often, sure, but not so often as to be a performance bottleneck I'd say. On the other hand the code today is, I'd say, safe.

Reply:
I agree with @niklasad1, we should just replace it with the safe code unless it introduces a perf bottleneck.